applications/FSIapplication/custom_utilities/partitioned_fsi_utilities.hpp

```diff
@@ -90,20 +91,61 @@ class PartitionedFSIUtilities
     */
-
-    /*@{ */
-    PartitionedFSIUtilities(const PartitionedFSIUtilities& Other);
-    /*@{ */
+    PartitionedFSIUtilities(const PartitionedFSIUtilities& Other) = delete;
 
     /** Destructor.
      */
-
-    virtual ~PartitionedFSIUtilities(){}
+    virtual ~PartitionedFSIUtilities() = default;
 
     /*@} */
     /**@name Public Operators*/
     /*@{ */
 
+    /**
+     * @brief Create a coupling element based skin object
+     * This method creates an element based skin model part by
+     * copying the conditions of a given skin model part
+     * @param rOriginInterfaceModelPart Origin skin model part to copy the conditions from
+     * @param rDestinationInterfaceModelPart Empty destination modelpart to create the skin elements
+     */
+    void CreateCouplingElementBasedSkin(
+        const ModelPart &rOriginInterfaceModelPart,
+        ModelPart &rDestinationInterfaceModelPart)
+    {
+        // Check the origin interface model part
+        KRATOS_ERROR_IF(rOriginInterfaceModelPart.NumberOfNodes() == 0) << "Origin model part has no nodes." << std::endl;
+        KRATOS_ERROR_IF(rOriginInterfaceModelPart.NumberOfConditions() == 0) << "Origin model part has no conditions." << std::endl;
+
+        // Check the destination interface model part
+        KRATOS_ERROR_IF(rDestinationInterfaceModelPart.IsSubModelPart()) << "Destination model part must be a root model part." << std::endl;
+        KRATOS_ERROR_IF(rDestinationInterfaceModelPart.NumberOfNodes() != 0) << "Destination interface model part should be empty. Current number of nodes: " << rDestinationInterfaceModelPart.NumberOfNodes() << std::endl;
+        KRATOS_ERROR_IF(rDestinationInterfaceModelPart.NumberOfElements() != 0) << "Destination interface model part should be empty. Current number of elements: " << rDestinationInterfaceModelPart.NumberOfElements() << std::endl;
+
+        // Emulate the origin interface nodes in the coupling skin
+        for (const auto &r_node : rOriginInterfaceModelPart.Nodes()) {
+            rDestinationInterfaceModelPart.CreateNewNode(r_node.Id(), r_node);
+        }
+
+        // Create the new element based skin
+        auto p_fake_prop = Kratos::make_shared<Properties>(0);
+        for (const auto &r_cond: rOriginInterfaceModelPart.Conditions()) {
+            // Set the points array
+            Geometry<Node<3>>::PointsArrayType points_array;
+            for (const auto &r_node : r_cond.GetGeometry()) {
+                points_array.push_back(rDestinationInterfaceModelPart.pGetNode(r_node.Id()));
+            }
+
+            // Set the element nodes vector
+            std::vector<ModelPart::IndexType> nodes_vect;
+            for (auto &r_node : r_cond.GetGeometry()) {
+                nodes_vect.push_back(r_node.Id());
+            }
+
+            // Create the new skin element
+            rDestinationInterfaceModelPart.CreateNewElement(this->GetSkinElementName(), r_cond.Id(), nodes_vect, p_fake_prop);
```

Comment:
Note that you are creating elements with a condition that is not known by the modelpart... Probably simpler to just retrieve Property(0) (it should be created if not there), to keep internal consistency of the modelpart

Reply:
I did that because the properties of these elements are not expected to be used. In any case, it probably makes more sense to use the same properties as the original condition.